core/src/main/scala/io/projectglow/bgen/BgenRowToInternalRowConverter.scala

```diff
@@ -23,14 +23,16 @@ import org.apache.spark.sql.types.{ArrayType, StructType}
 import org.apache.spark.unsafe.types.UTF8String
 
 import io.projectglow.common.{BgenGenotype, BgenRow, GlowLogging, VariantSchemas}
+import io.projectglow.sql.expressions.HardCalls
 import io.projectglow.sql.util.RowConverter
 
 /**
  * Converts [[BgenRow]]s into [[InternalRow]] with a given required schema. During construction,
  * this class will throw an [[IllegalArgumentException]] if any of the fields in the required
  * schema cannot be derived from a BGEN record.
  */
-class BgenRowToInternalRowConverter(schema: StructType) extends GlowLogging {
+class BgenRowToInternalRowConverter(schema: StructType, hardCallsThreshold: Option[Double])
```

Comment:
Why is this an option? Seems like we could always pass in the default or the user-provided override.